### PR TITLE
fix(inline-notification): update close btn focus outline & action btn specificity

### LINKED ISSUE #4561: [Inline Notification] can not focus on the close icon using keyboard
## What package(s) are you using?

- [x] `carbon-components` 10.7.3
- [x] `carbon-components-react` 7.7.3
- [x] `@carbon/icons-react` 10.6.0

## Detailed description

Can not focus on the close icon using keyboard

> Is this issue related to a specific component?

Yes, Inline Notification

> What did you expect to happen? What happened instead? What would you like to
> see changed?

I would expect to be able to focus on the close icon using keyboard and be able to close notification using keyboard.

> What browser are you working in?

Google Chrome Version 78.0.3904.70 (Official Build) (64-bit)

> What version of the Carbon Design System are you using?

10

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

Hyper Protect, release ~ Q4

## Steps to reproduce the issue

1. Use 'tab' key to navigate
2. Try to focus on the close icon  using keyboard

> Please create a reduced test case in CodeSandbox

https://codesandbox.io/s/codesandbox-ep4lz

### LINKED ISSUE #4567: [Inline Notification] action button style regression
## What package(s) are you using?

- [x] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

Currently there's a style regression for the "action" button in the `InlineNotification` --

http://react.carbondesignsystem.com/?path=/story/notifications--inline

Screen Shot 2019-11-05 at 10 53 06 AM

(The action button is misaligned & the text is the default link color, not the inverse link color that is accessible against the inverse background)

It looks like the action button styles are being overridden, because they are no longer specific enough with recent changes, perhaps from #4299?

### PR DESCRIPTION
**Pull request by jendowns**
Closes #4561
Closes #4567 

#### Changelog

**New**

- add focus state for ~close~ all buttons in `InlineNotification` & `ToastNotification`

**Changed**

- add a level of specificity to the action button styles in the `InlineNotification` so they aren't overridden by recent updates
- adjusted the svg position of the close button icon for the `ToastNotification`, as it looked a little too high 🤔

### CODE REVIEW

**Review by abbeyhrt — APPROVED** [review]
This looks good to me! I'm gonna request a review from the designers to make sure the focus outline is good on their end 👍

**Review by aagonzales — COMMENTED** [review]
The high contrast notification should actually be using `$inverse-focus-ui` so it will have enough contrast.

**Comment by jendowns** [comment]
@aagonzales So the focus outline of the close button for `InlineNotification` & `ToastNotification`, plus the action button focus outline in the `InlineNotification`, should be `$inverse-focus-ui` when in ~low~ high contrast mode?

**Comment by aagonzales** [comment]
Yes, exactly (sorry for the vagueness)

**Comment by jendowns** [comment]
Thank you @aagonzales! I just pushed an update that changes the outline color for all of those buttons

I also happened to notice that the `ToastNotification`'s close button's icon was a little off? 🤔 I added a couple rules to fix that. 

Here's what it looked like *before* my change (the icon seems a little too high):
Screen Shot 2019-11-05 at 2 43 57 PM

File: packages/components/src/components/notification/_inline-notification.scss [hunk 1]
```diff
@@ -177,6 +177,12 @@
       color: $inverse-link;
     }
 
+    &:focus {
+      border-color: transparent;
+      outline: 2px solid $inverse-focus-ui;
+      outline-offset: -2px;
```
Comment (jendowns) [inline packages/components/src/components/notification/_inline-notification.scss:183]: With this I made a judgement call, so let me know what you think! I couldn't find an instance of an inverse ghost button, and in this case the border + outline colors were strange when one or the other was `$inverse-focus-ui`. So this makes the focus outline of the button one color, similar to the close buttons.

**Review by aagonzales — APPROVED** [review]
You're awesome! ⭐️ That all good good to me!

**Pull request merged**